packages/js/src/Verto.ts

```diff
@@ -135,4 +135,15 @@ export default class Verto extends BrowserSession {
   protected _onSocketMessage(msg: any) {
     VertoHandler(this, msg)
   }
+
+  protected _onSocketCloseOrError(event: any): void {
```

Comment:
What is valid reason?

Reply:
The problem is that on the base class `BaseSession.ts` a `connect()` will be fired on both cases:
```Typescript
if (this._autoReconnect) {
      this._reconnectTimeout = setTimeout(() => this.connect(), this.reconnectDelay)
    }
``` 
If we only set it at `onClose` we still have the `OnError` not handled.

And if I'm not mistaken the onError could be fired first if you are trying to write on a socket in a "BROKEN_PIPE" status. It's disconnected but we are writing on it before handling the `onClose`

Reply:
> What is valid reason? And what reason is not valid? How can we determine the need for `needResume` to be set?

@andywolk By looking at the code... `needResume` is a flag that will be sent to the server on connect.
IMO `needResume` needs to be set on every reconnect case and that includes both onError and onClose since both start a reconnection timer.

Reply:
> > What is valid reason? And what reason is not valid? How can we determine the need for `needResume` to be set?
> 
> @andywolk By looking at the code... `needResume` is a flag that will be sent to the server on connect. IMO `needResume` needs to be set on every reconnect case and that includes both onError and onClose since both start a reconnection timer.

Thank you. That makes sense.